mk/tap.mk

```diff
@@ -0,0 +1,9 @@
+AM_TESTS_ENVIRONMENT= \
+	G_TEST_SRCDIR="$(abs_srcdir)" 		\
+	G_TEST_BUILDDIR="$(abs_builddir)" 	\
+	G_DEBUG=gc-friendly 			\
```

Comment:
Do we still need the G_ variables?

Reply:
Ah, I was hoping you'd comment on that.  Do we really need any of it?  AM_TESTS_ENVIRONMENT is environment variables that will get set up for running the tests.  We could put whatever we want in here.

Reply:
The G_TEST_* variables were for glib unit tests, so we can drop those.

G_DEBUG and MALLOC_PERTURB_ are similar, I guess G_DEBUG would apply when glib doesn't use malloc(), but it seems they are kind of in opposition (G_DEBUG sets freed memory to 0, MALLOC_PERTURB_ sets it to the given value). I suppose they're worthwhile, though I've never noticed them catch anything.

MALLOC_CHECK_ is probably worthwhile.

Reply:
Fixed.  I wonder if it would be worth renaming this to test.mk (or check.mk, to hint that it's for "make check" and not for CTS) and then looking around to see if there's any other common code that could be merged into here.

Reply:
Definitions of LDADD and AM_CPPFLAGS come to mind.

Reply:
I don't think we could move much here since eventually all the libraries will have tests, and it'll vary. The name is fine, or check.mk or unit-tests.mk would also work.